src/robot/dial.ts

```diff
@@ -185,3 +193,50 @@ export const createRobotClient = async (
 
   return client;
 };
+
+/**
+ * Get a Viam Transport Factory after getting the accessToken.
+ *
+ * In dialOpts.credentials, the credential type cannot be a robot secret. The
+ * credential type to use would preferably be the organization api key.
+ */
+export const createViamTransportFactory = async (
+  serviceHost: string,
+  dialOpts: DialOptions
+): Promise<grpc.TransportFactory> => {
+  if (!dialOpts.credentials) {
+    throw new Error(`credential cannot be none`);
+  } else if (dialOpts.credentials.type === 'robot-secret') {
+    throw new Error(`credential type cannot be 'robot secret'`);
+  } else if (!dialOpts.authEntity) {
+    throw new Error(`auth entity cannot be None`);
```

Comment:
`None` is not a value in JS/TS, so a more helpful message might be:

```suggestion
    throw new Error(`auth entity cannot be null, undefined, or an empty value.`);
```